modules/tiles/src/tileset/traversers/i3s-tile-manager.ts

```diff
@@ -7,33 +10,44 @@ const STATUS = {
 // A helper class to manage tile metadata fetching
 export default class I3STileManager {
   private _statusMap: object;
+  private frameCounter = new FrameCounter();
 
   constructor() {
     this._statusMap = {};
   }
 
-  add(request, key, callback, frameState) {
+  add(request, key, callback, frameState: FrameState) {
     if (!this._statusMap[key]) {
+      const {frameNumber} = frameState;
       this._statusMap[key] = {request, callback, key, frameState, status: STATUS.REQUESTED};
+      this.frameCounter.increase(frameNumber);
       request()
         .then((data) => {
           this._statusMap[key].status = STATUS.COMPLETED;
+          this.frameCounter.decrease(frameNumber);
```

Comment:
Please add comments, what do we gain by adding arbitrary frame numbers to a counter (as opposed to just incremnting and decrementing it?